pkg/plugins/manager/loader/assetpath/assetpath.go

```diff
@@ -36,7 +40,8 @@ func (s *Service) Base(pluginJSON plugins.JSONData, class plugins.Class, pluginD
 // Module returns the module.js path for the specified plugin.
 func (s *Service) Module(pluginJSON plugins.JSONData, class plugins.Class, pluginDir string) (string, error) {
 	if class == plugins.ClassCore {
-		return path.Join("app/plugins", string(pluginJSON.Type), filepath.Base(pluginDir), "module"), nil
+		baseDir := getBaseDir(pluginDir, false)
+		return path.Join("app/plugins", string(pluginJSON.Type), baseDir, "module"), nil
```

Comment:
here, I am keeping the module URL as `app/plugins/datasource/testdata/module` to differentiate the path from the one used as a external plugin (`plugins/testdata`) since the path to core plugins is hardcoded: https://github.com/grafana/grafana/blob/main/public/app/features/plugins/built_in_plugins.ts#L100